packages/metro-react-native-babel-preset/src/configs/main.js

```diff
@@ -20,6 +20,15 @@ function isTSXSource(fileName) {
   return !!fileName && fileName.endsWith('.tsx');
 }
 
+function notTypeScriptSource(fileName) {
+  return !isTypeScriptSource(fileName) && !isTSXSource(fileName);
+}
+
+function optionsForTransformTS(opts) {
+  // `allowDeclareFields` will be enabled by default in Babel 8
+  return {...opts, allowDeclareFields: true, allowNamespaces: true};
+}
+
 const defaultPluginsBeforeRegenerator = [
   [require('@babel/plugin-syntax-flow')],
```

Comment:
Should we also make this conditional on `notTypeScriptSource`? (e.g. move it into the same `plugins` block as `flow-strip-type`)

Reply:
Yes, that's probably a good idea.

Reply:
Actually I wonder if the plugin is even necessary? Since the `transform-flow-strip-types` already includes it for parsing flow syntax, and that plugin is earlier in the configuration, this plugin is probably not actually doing anything.